pennylane_lightning/lightning_qubit/_adjoint_jacobian.py

```diff
@@ -0,0 +1,362 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Internal methods for adjoint Jacobian differentiation method.
+"""
+from os import getenv
+from typing import List
+from warnings import warn
+
+import numpy as np
+import pennylane as qml
+from pennylane import BasisState, Projector, QuantumFunctionError, Rot, StatePrep
+from pennylane.measurements import Expectation, MeasurementProcess, State
+from pennylane.operation import Operation, Tensor
+from pennylane.tape import QuantumTape
+
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+from pennylane_lightning.core.lightning_base import _chunk_iterable
+
+# pylint: disable=import-error, no-name-in-module, ungrouped-imports
+try:
+    from pennylane_lightning.lightning_qubit_ops.algorithms import (
+        AdjointJacobianC64,
+        AdjointJacobianC128,
+        create_ops_listC64,
+        create_ops_listC128,
+    )
+except ImportError:
+    pass
+
+from ._state_vector import LightningStateVector
+
+
+class LightningAdjointJacobian:
+    """Check and execute the adjoint Jacobian differentiation method.
+
+    Args:
+        qubit_state(LightningStateVector): State Vector to calculate the adjoint Jacobian with.
+        batch_obs(bool): If serialized tape is to be batched or not.
+    """
+
+    def __init__(self, qubit_state: LightningStateVector, batch_obs: bool = False) -> None:
+        self._qubit_state = qubit_state
+        self._state = qubit_state.state_vector
+        self._dtype = qubit_state.dtype
+        self._jacobian_lightning = (
+            AdjointJacobianC64() if self._dtype == np.complex64 else AdjointJacobianC128()
+        )
+        self._create_ops_list_lightning = (
+            create_ops_listC64 if self._dtype == np.complex64 else create_ops_listC128
+        )
+        self._batch_obs = batch_obs
+
+    @property
+    def qubit_state(self):
+        """Returns a handle to the LightningStateVector class."""
+        return self._qubit_state
+
+    @property
+    def state(self):
+        """Returns a handle to the Lightning internal data class."""
+        return self._state
+
+    @property
+    def dtype(self):
+        """Returns the simulation data type."""
+        return self._dtype
+
+    @staticmethod
+    def _check_adjdiff_supported_measurements(
```

Comment:
So all these validation checks should get run in `LightningQubit.preprocess`.  So the question is whether or not we want to "re-validate" these again, or just trust that someone actually ran preprocessing.

Reply:
Removed!